scripts/migrations/2020-08-update_migration_created_date.py

```diff
@@ -0,0 +1,83 @@
+dynamodb = get_aws_resource("dynamodb", role_arn=READ_ONLY_ROLE_ARN)
+s3 = get_aws_resource("s3", role_arn=READ_ONLY_ROLE_ARN)
+
+vhs_table = "vhs-storage-staging-manifests-2020-07-24"
+
+backfill_vhs_table = "vhs-storage-staging-manifests-2020-08-19"
+
+
+def get_bucket_key(item):
+    try:
+        bucket = item["payload"]["namespace"]
+        key = item["payload"]["path"]
+    except KeyError:
+        try:
+            bucket = item["payload"]["bucket"]
+            key = item["payload"]["key"]
+        except:
+            raise RuntimeError(f"Find s3 bucket, key in {item}")
+        else:
+            return bucket, key
+    else:
+        return bucket, key
+
+
+def get_vhs_json(bucket, key):
+    return json.loads(s3.Object(bucket, key).get()["Body"].read().decode("utf-8"))
+
+
+def put_vhs_json(bucket, key, content):
+    try:
+        s3object = s3.Object(bucket, key)
+        # s3object.put(
+        #     Body=(bytes(json.dumps(content).encode('UTF-8')))
+        # )
+    except ClientError as e:
+        raise RuntimeError(f"Error updating backfill vhs object s3://{bucket}/{key}",e)
+    except:
+        raise RuntimeError(f"Unknown error when updating backfill vhs object s3://{bucket}/{key}")
+
+
+for item in scan_table(TableName=vhs_table):
+    id = item["id"]
+    version = item["version"]
+    bucket, key = get_bucket_key(item)
+    try:
+        vhs_content = get_vhs_json(bucket, key)
+    except:
+        raise RuntimeError(f"Cannot read s3 entry for {id}: {item}")
+    else:
+        created_date = vhs_content["createdDate"]
+        try:
+            backfilled_item = dynamodb.Table(backfill_vhs_table).get_item(
+                Key={"id": id, "version": version}
+            )["Item"]
+        except KeyError:
+            print(f"Cannot find backfill storage manifest for {id}:{version}!!!!")
+        else:
+            backfilled_bucket = backfilled_item["payload"]["bucket"]
+            backfilled_key = backfilled_item["payload"]["key"]
+            backfilled_json = get_vhs_json(backfilled_bucket, backfilled_key)
+
+            diff = DeepDiff(vhs_content, backfilled_json, ignore_order=True)
+
+            if diff:
+                values_changed = diff.pop("values_changed", None)
+                items_added = diff.pop("iterable_item_added", None)
+                if values_changed and values_changed.keys()-["root['createdDate']"]:
+                    raise RuntimeError(f"Unexpected values changed in {values_changed} for {id}:{version}!")
+                if items_added and items_added.keys() - ["root['replicaLocations'][1]"]:
+                    raise RuntimeError(f"Unexpected values changed in {items_added} for {id}:{version}!")
+                if diff:
+                    raise RuntimeError(f"Unexpected values changed in {diff} for {id}:{version}!")
+
+            backfilled_json["createdDate"] = created_date
+            put_vhs_json(backfilled_bucket, backfilled_key, backfilled_json)
```

Comment:
This is still overwriting the existing object, right?

Reply:
Also… hmm, is the backfill bucket the appropriate place? It's got backfill in the name, which feels a bit icky. Maybe we should be pushing back into the live VHS (or its bucket)? 😕

Reply:
yes, I still have to change that...the syntax to update something in dynamo isn't incredibly friendly to me so trying to navigate that right now. 

Yeh, maybe if we are updating the VHS record anyway, it could be better updating the old one instead of the new one